cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -347,8 +347,13 @@ public void received(StreamWriterHeartbeat heartbeat, NotificationContext notifi
         }, heartbeatsSubscriptionExecutor);
       } catch (NotificationFeedException e) {
         // Most probably, the dataset service is not up. We retry
-        LOG.warn("Could not perform operation on HeartbeatsFeed. Retrying.", e);
-        TimeUnit.MILLISECONDS.sleep(1000);
+        LOG.warn("Could not perform operation on HeartbeatsFeed. Retrying in one second.");
```

Comment:
Minor nit: Move it to a separate method, since the code is the same here and below?